example/android/build.gradle

```diff
@@ -24,3 +24,41 @@ task fetchDependencies() {
         }
     }
 }
+
+allprojects {
+    repositories {
+        maven {
+            url("$rootDir/../../node_modules/detox/Detox-android")
+        }
+    }
+    afterEvaluate { project ->
+        def androidExtension = project.extensions.findByName('android')
+
+        if(androidExtension != null && project.name == 'app') {
+            // android extension from test-app pacakge
+            androidExtension.defaultConfig {
+                ndk {
+                    abiFilters 'arm64-v8a', 'x86', 'x86_64'
+                }
+            }
+
+            androidExtension.signingConfigs {
+                test {
+                    storeFile file('debug.keystore')
+                    storePassword 'android'
+                    keyAlias 'androiddebugkey'
+                    keyPassword 'android'
+                }
+            }
+
+            androidExtension.buildTypes.release.signingConfig = androidExtension.signingConfigs.test
+            androidExtension.testBuildType = 'release'
+
+            androidExtension.sourceSets.androidTest.java.srcDirs += "$rootDir/app/src/androidTest/java"
+
+            project.dependencies {
+                androidTestImplementation('com.wix:detox:+')
+            }
+        }
+    }
```

Comment:
@arazabishov: Do you think we can simplify how devs can customize `react-native-test-app`? Not sure if this is normal as far as Gradle configs go but it seems a bit cumbersome to add `signingConfigs` and Detox specific things, imo.

Reply:
I think it is fine to leverage the flexibility of Gradle when needed, like in this case. If we were about to parameterize all of the things above, we will find ourselves trying to replicate AGP's API. What we can do is provide a convenient way to tap into androidExtension, so users don't have to search for it themselves.

Reply:
If this is normal, then I think it's probably fine. We can add a "recipe" in our wiki and point people in that direction when they need to add stuff.